squire_core/src/accounts.rs

```diff
@@ -32,13 +35,112 @@ pub fn all_users() -> GetAllUsersResponse {
     GetAllUsersResponse::new(map)
 }
 
+#[get("/users/get/<id>/permissions")]
+pub fn user_permissions(id: UserAccountId) -> GetUserPermissionsResponse {
+    GetUserPermissionsResponse::new(
+        ORGS_MAP
+        .get()
+        .unwrap()
+        .get(&UserAccountId(id))
+        .map(|user| user.get_current_permissions()),
+    )
+}
+
+#[post("/users/update/<id>")]
+pub fn update_user_account(id: UserAccountId, 
+    user_name: Option<String>, 
+    display_name: Option<String>,
+    delete_user_name: Option<Boolean>,
+    delete_display_name: Option<Boolean>
+) -> UpdateSquireAccountResponse {
+    UpdateSquireAccountResponse::new(
+        if(user_name.unwrap() != None) {
```

Comment:
I understand what you're trying to do, but this statement makes very little techincal sense. `Option`s are how we represent something that is/isn't there. Because of this, we can't treat, say, an `Option<String>` as a `String`. We have to account for what happens then the value is `None` (i.e. there is no `String`). The `unwrap` method is shorthand for saying "I know this value exists". Calling `unwrap` on a `None` will cause a panic (causing the method, and perhaps the program, to end).

`Option`s are part of Rust's algebraic type system. They are super powerful. I recommend you read up on them some more in the[ Rust book](https://doc.rust-lang.org/book/ch06-01-defining-an-enum.html). 

You can destructure data in Rust very easily. I believe you're trying to say "if this username isn't empty (i.e. `Some`) then we want to perform an operation". That's easily done like so
```rust
if let Some(name) = user_name { 
  /* code block */
}
```